src/server/db_slice.cc

```diff
@@ -1515,7 +1515,10 @@ void DbSlice::CallChangeCallbacks(DbIndex id, std::string_view key, const Change
 
   DVLOG(2) << "Running callbacks for key " << key << " in dbid " << id;
   FetchedItemsRestorer fetched_restorer(&fetched_items_);
-  std::unique_lock<LocalBlockingCounter> lk(block_counter_);
+  // Do not use std::scoped_lock because we need to aqcuire the logical_lk
+  // regardless since it merely increments a counter and does not block.
+  std::unique_lock logical_lk(block_counter_);
```

Comment:
Sorry, I still don't understand.
If we remove `block_counter_` entirely, and instead lock `local_mu_` in `UnregisterOnChange()`, wouldn't that be ok? Why not?

Reply:
oh that's what you meant, yes I think this should be ok

Reply:
then lets do that :)